### PR TITLE
ROS2: Remove jinja; add CI; framestore working

### PR DESCRIPTION
**Pull request by marinagmoreira**

### CODE REVIEW

File: communications/ff_msgs/action/Motion.action [hunk 1]
```diff
@@ -111,9 +111,8 @@ ff_msgs/ControlState[] segment
 ff_msgs/MotionState state
 
 # Control progress
-{% if ROS1 %}
-ff_msgs/ControlFeedback progress                    # TODO(@trey0): ROS2 does not allow this
-{% endif %}
+# ff_msgs/ControlFeedback progress                    # ROS2 does not allow this
```
Comment (kbrowne15) [inline communications/ff_msgs/action/Motion.action:114]: Is this a todo for later? Seems like we may need it..

File: shared/ff_util/include/ff_util/ff_component.h [hunk 2]
```diff
@@ -70,12 +70,16 @@ class FreeFlyerComponent {
...
 
-  // Use default name from freeflyer
-  // COMPOSITION_PUBLIC
-  explicit FreeFlyerComponent(const rclcpp::NodeOptions & options, bool autostart_hb_timer = true);
+  // Use default name from freeflyer - crashes in ROS2
+  // explicit FreeFlyerComponent(const rclcpp::NodeOptions & options, bool autostart_hb_timer = true);
   // Explicitly specift the name
   explicit FreeFlyerComponent(const rclcpp::NodeOptions& options, std::string const& name,
                             bool autostart_hb_timer = true);
+  // Explicitly specift the node from gazebo
```
Comment (kbrowne15) [inline shared/ff_util/include/ff_util/ff_component.h:78]: specift --> specify

File: shared/ff_util/src/ff_component/ff_component.cc [hunk 3]
```diff
@@ -18,37 +18,48 @@
...
   sleeping_(false),
   heartbeat_queue_size_(5),
-  node_name_(node) {
+  node_name_(name) {
+  Setup(name);
 }
 
-FreeFlyerComponent::FreeFlyerComponent(const rclcpp::NodeOptions & options, bool autostart_hb_timer) :
-  node_(std::make_shared<rclcpp::Node>("", options)),
+// FreeFlyerComponent::FreeFlyerComponent(const rclcpp::NodeOptions & options, bool autostart_hb_timer) :
```
Comment (kbrowne15) [inline shared/ff_util/src/ff_component/ff_component.cc:38]: I think you can remove this constructor.

File: shared/msg_conversions/CMakeLists.txt [hunk 4]
```diff
@@ -76,57 +45,49 @@ endif()
...
-  ${catkin_INCLUDE_DIRS}         # ROS1
   ${config_reader_INCLUDE_DIRS}  # ROS2
 )
 
 # Declare C++ libraries
 add_library(msg_conversions
   src/msg_conversions.cc
 )
-target_link_libraries(msg_conversions ${EIGEN_LIBRARIES} ${catkin_LIBRARIES})
+# target_link_libraries(msg_conversions ${EIGEN_LIBRARIES} ${catkin_LIBRARIES})
```
Comment (kbrowne15) [inline shared/msg_conversions/CMakeLists.txt:55]: You can remove this.

**Review by kbrowne15 — APPROVED** [review]

File: communications/ff_msgs/action/Motion.action (hunk 1 above)
Comment (marinagmoreira) [reply communications/ff_msgs/action/Motion.action:114]: The split between goal/feedback/result is not present anymore on ROS2, only a full action message. This is still a TODO that we need to either create a separate feedback message type or pass in the entire action message and only fill in the feedback portion. I'll add this back as TODO, thank you for noticing!

**Pull request merged**